source/adapters/hip/device.cpp

```diff
@@ -821,7 +829,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
   case UR_DEVICE_INFO_GPU_HW_THREADS_PER_EU:
   case UR_DEVICE_INFO_MAX_MEMORY_BANDWIDTH:
   case UR_DEVICE_INFO_BFLOAT16:
-    return UR_RESULT_ERROR_INVALID_ENUMERATION;
+  case UR_DEVICE_INFO_IL_VERSION:
+  case UR_DEVICE_INFO_ASYNC_BARRIER:
+  case UR_DEVICE_INFO_VIRTUAL_MEMORY_SUPPORT:
+    return UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION;
```

Comment:
So is this why the test was failing, you needed to move 
```
return UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION;
```

further down ?

Reply:
Yep, return `UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION` instead of `UR_RESULT_ERROR_INVALID_ENUMERATION`.